src/bin/pg_autoctl/cli_create_drop_node.c

```diff
@@ -970,13 +999,74 @@ cli_drop_node_getopts(int argc, char **argv)
 		(!IS_EMPTY_STRING_BUFFER(options.hostname) ||
 		 options.pgSetup.pgport != 0))
 	{
-		log_error("Please use either --hostname and --pgport or ---destroy");
+		log_error("Please use either --hostname and --pgport or --destroy");
 		log_info("Destroying a node is not supported from a distance");
 		exit(EXIT_CODE_BAD_ARGS);
 	}
 
 	/* now that we have the command line parameters, prepare the options */
-	(void) prepare_keeper_options(&options);
+	/* when we have a monitor URI we don't need PGDATA */
+	if (cli_use_monitor_option(&options))
+	{
+		if (!IS_EMPTY_STRING_BUFFER(options.pgSetup.pgdata))
+		{
+			log_warn("Given --monitor URI, the --pgdata option is ignored");
+			log_info("Connecting to monitor at \"%s\"", options.monitor_pguri);
+
+			/* the rest of the program needs pgdata actually empty */
+			bzero((void *) options.pgSetup.pgdata,
+				  sizeof(options.pgSetup.pgdata));
+		}
+	}
+	else
+	{
+		(void) prepare_keeper_options(&options);
+	}
+
+	/*
+	 * pg_autoctl drop node can be used with one of those set of arguments:
+	 *   --pgdata ...                 # to drop the local node
+	 *   --formation ... --name ...   # address a node on the monitor
+	 *   --hostname ... --port ...    # address a node on the monitor
+	 */
+	if (!IS_EMPTY_STRING_BUFFER(options.pgSetup.pgdata))
+	{
+		if (!IS_EMPTY_STRING_BUFFER(options.formation) ||
+			!IS_EMPTY_STRING_BUFFER(options.name) ||
+			!IS_EMPTY_STRING_BUFFER(options.hostname) ||
+			options.pgSetup.pgport != 0)
+		{
+			log_fatal("pg_autoctl drop node can either drop the local node "
+					  "when given PGDATA or --pgdata, or drop another node "
+					  "from the monitor, using --formation and --name, or "
+					  "using --hostname and --pgport.");
```

Comment:
I also fixed optargs checking and simplified it.

Reply:
> A typo or a copy/past'o in the --name argument and you're dropping the wrong one because you're not connected (ssh'ed) into the node you though you were, or maybe you're using the wrong terminal tab. That's why I think if we have PGDATA or --pgdata, that should be pointed to the node we want to drop.

I think that's a fair point, but I think the best solution would be to have a confirmation prompt like:
```
You are going to drop a different node then you are currently connected to. Please type the name of that node again:
```

However, I don't think this is necessary for this PR.

Reply:
I'm not a fan of interactive prompting in tooling: it makes scripting unnecessarily complicated. We can still open an issue about this and discuss how to best solve it. In that case, is this PR ready then?

Reply:
Yes, I'm fine with not allowing this with this PR. However it seems you removed a little bit too many error checks in your last changes. Right now this command succeeds:
```
pg_autoctl drop node --pgdata node1 --name node2
```
But it removes node1 instead of node2. Can you fix that, by throwing an error.